lib/foreman_inventory_upload/async/upload_report_job.rb

```diff
@@ -37,7 +37,9 @@ def env
       end
 
       def http_proxy_string
-        @http_proxy_string ||= HttpProxy.default_global_content_proxy&.full_url
+        @http_proxy_string ||=
+          HttpProxy.default_global_content_proxy&.full_url ||
```

Comment:
If this is trying to address the http proxy situation on 6.6 (katello 3.12ish), then `HttpProxy.default_global_content_proxy` is not defined and calling it would raise a `NoMethodError`, no?

Reply:
The method is implemented a bit differently in foreman 22 compatible version: https://github.com/theforeman/foreman_rh_cloud/blob/foreman_1_22/lib/foreman_inventory_upload/async/upload_report_job.rb#L41 . We will change the method accordingly once we cherry-pick this PR to 22 branch. The idea is to leave the code intact: first check `global_content_proxy`, if it's not there - fall back to `cdn_proxy` setting.

Reply:
Make sense, thanks for the explanation